x/ibc/07-tendermint/misbehaviour.go

```diff
@@ -66,7 +67,18 @@ func checkMisbehaviour(
 	infractionHeight := evidence.GetHeight()
 	infractionTime := evidence.GetTime()
 	ageDuration := currentTimestamp.Sub(infractionTime)
-	ageBlocks := height - uint64(infractionHeight)
+	ageBlocks := uint64(infractionHeight) - height
+
+	// assert that trustedVals is NextValidators of last trusted header
+	// to do this, we check that trustedVals.Hash() == consState.NextValidatorsHash
+	tvalHash := evidence.Header1.TrustedValidators.Hash()
```

Comment:
we should do this for both headers if we are not requiring the header trusted heights to be equal

Reply:
++